### PR TITLE
feat(ui): Add animation to TabSelectionIndicator

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Do we want this?

### CODE REVIEW

File: static/app/components/tabs/tab.tsx [hunk 1]
```diff
@@ -176,15 +173,11 @@ const TabSelectionIndicator = styled('div')<{
         height: 3px;
 
         bottom: 0;
-        left: 50%;
-        transform: translateX(-50%);
```
Comment (evanpurkhiser) [inline static/app/components/tabs/tab.tsx:180]: It actually seems like we didn't need these at all afaict

File: static/app/components/tabs/tab.tsx [hunk 2]
```diff
@@ -47,7 +48,13 @@ function BaseTab(
       <HoverLayer orientation={orientation} />
       <FocusLayer orientation={orientation} />
       {rendered}
-      <TabSelectionIndicator orientation={orientation} selected={isSelected} />
+      {isSelected && (
+        <TabSelectionIndicator
+          layoutId="indicator"
+          transition={{type: 'spring', stiffness: 250, damping: 25}}
```
Comment (vuluongj20) [inline static/app/components/tabs/tab.tsx:54]: I like how this makes the animation feel more speedy, but as you said the spring may affect the performance. Could we remove the spring effect (where the indicator overshoots and then bounces back) but still make it feel as speedy?

Comment (evanpurkhiser) [reply static/app/components/tabs/tab.tsx:54]: yeah let me play with it a bit more later

File: static/app/components/tabs/tab.tsx [hunk 3]
```diff
@@ -155,19 +162,13 @@ const FocusLayer = styled('div')<{orientation: Orientation}>`
...
 }>`
   position: absolute;
   border-radius: 2px;
   pointer-events: none;
-  background: ${p => (p.selected ? p.theme.active : 'transparent')};
-  transition: background 0.1s ease-out;
-
-  li[aria-disabled='true'] & {
-    background: ${p => (p.selected ? p.theme.subText : 'transparent')};
-  }
```
[the remark this answers is not in the document]
Comment (evanpurkhiser) [reply static/app/components/tabs/tab.tsx:170]: oh I see why this is here

**Comment by evanpurkhiser** [comment]
Closing for now.

**Pull request closed without merging**